internal/file/file_manager_service.go

```diff
@@ -792,3 +829,222 @@ func tempBackupFilePath(fileName string) string {
+	content []byte,
+	path string,
+	permission string,
+) error {
+	writeErr := fms.fileOperator.Write(
+		ctx,
+		content,
+		path,
+		permission,
+	)
+
+	if writeErr != nil {
+		return fmt.Errorf("failed to write downloaded content to temp file %s: %w", path, writeErr)
+	}
+
+	return nil
+}
+
+// downloadFileContent performs an HTTP GET request to the given URL and returns the file content as a byte slice.
+func (fms *FileManagerService) downloadFileContent(
+	ctx context.Context,
+	file *mpi.File,
+) (content []byte, headers DownloadHeader, err error) {
+	fileName := file.GetFileMeta().GetName()
+	downloadURL := file.GetExternalDataSource().GetLocation()
+	externalConfig := fms.agentConfig.ExternalDataSource
+
+	if !isDomainAllowed(downloadURL, externalConfig.AllowedDomains) {
+		return nil, DownloadHeader{}, fmt.Errorf("download URL %s is not in the allowed domains list", downloadURL)
+	}
+
+	httpClient, err := fms.setupHTTPClient(ctx, externalConfig.ProxyURL.URL)
+	if err != nil {
+		return nil, DownloadHeader{}, err
+	}
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, downloadURL, nil)
+	if err != nil {
+		return nil, DownloadHeader{}, fmt.Errorf("failed to create request for %s: %w", downloadURL, err)
+	}
+
+	if externalConfig.ProxyURL.URL != "" {
+		fms.addConditionalHeaders(ctx, req, fileName)
+	} else {
+		slog.DebugContext(ctx, "No proxy configured; sending plain HTTP request without caching headers.")
+	}
+
+	resp, err := httpClient.Do(req)
+	if err != nil {
+		return nil, DownloadHeader{}, fmt.Errorf("failed to execute download request for %s: %w", downloadURL, err)
+	}
+	defer resp.Body.Close()
+
+	switch resp.StatusCode {
+	case http.StatusOK:
+		headers.ETag = resp.Header.Get("ETag")
+		headers.LastModified = resp.Header.Get("Last-Modified")
+	case http.StatusNotModified:
+		slog.DebugContext(ctx, "File content unchanged (304 Not Modified)", "file_name", fileName)
+		return nil, DownloadHeader{}, nil
+	default:
+		return nil, DownloadHeader{}, fmt.Errorf("download failed with status code %d", resp.StatusCode)
+	}
+
+	reader := io.Reader(resp.Body)
+	if fms.agentConfig.ExternalDataSource.MaxBytes > 0 {
+		reader = io.LimitReader(resp.Body, fms.agentConfig.ExternalDataSource.MaxBytes)
+	}
+
+	content, err = io.ReadAll(reader)
+	if err != nil {
+		return nil, DownloadHeader{}, fmt.Errorf("failed to read content from response body: %w", err)
+	}
+
+	slog.InfoContext(ctx, "Successfully downloaded file content", "file_name", fileName, "size", len(content))
+
+	return content, headers, nil
+}
+
+func isDomainAllowed(downloadURL string, allowedDomains []string) bool {
+	u, err := url.Parse(downloadURL)
+	if err != nil {
+		slog.Debug("Failed to parse download URL for domain check", "url", downloadURL, "error", err)
+		return false
+	}
+
+	hostname := u.Hostname()
+	if hostname == "" {
+		return false
+	}
+
+	for _, domain := range allowedDomains {
+		if domain == "" {
+			continue
+		}
+
+		if domain == hostname {
```

Comment:
Would the below work to reduce the number of if statements 
```suggestion
		if domain == hostname || sMatchesWildcardDomain(hostname, domain) {
```